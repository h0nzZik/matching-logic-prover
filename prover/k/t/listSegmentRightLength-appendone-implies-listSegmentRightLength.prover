claim \implies( \and( listSegmentRightLength( variable("H") { ArrayIntInt }
                                            , variable("X") { Int } 
                                            , variable("Y") { Int } 
                                            , variable("FA") { Set } 
                                            , variable("LA") { Int }
                                            , .Patterns
                                            ) 
                    , \equals(variable("F") { Set }, union(variable("FA") { Set }, singleton(variable("Y") { Int })))
                    , disjoint(variable("FA") { Set }, singleton(variable("Y") { Int }))
                    , \equals(variable("Z") { Int }, select(variable("H") { ArrayIntInt }, variable("Y") { Int }))
                    , \equals(variable("LENGTH") { Int }, plus(variable("LA") { Int }, 1))
                    , gt(variable("Y") { Int }, 0)
                    , .Patterns
                    )
              , \and( listSegmentRightLength( variable("H") { ArrayIntInt } 
                                            , variable("X") { Int } 
                                            , variable("Z") { Int } 
                                            , variable("F") { Set } 
                                            , variable("LENGTH") { Int }
                                            , .Patterns
                                            )
                   , .Patterns
                   )
              ) 
strategy search-bound(3)

/*
strategy right-unfold-Nth(0, 1) ; simplify ; direct-proof
*/
