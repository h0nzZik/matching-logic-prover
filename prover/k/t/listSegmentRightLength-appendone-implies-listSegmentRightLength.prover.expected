<prover>
  <goal>
    <id>
      0
    </id>
    <active>
      true
    </active>
    <parent>
      -1
    </parent>
    <k>
      \implies ( \and ( listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" ) { Int } , variable ( "FA" ) { Set } , variable ( "LA" ) { Int } , .Patterns ) , \equals ( variable ( "F" ) { Set } , union ( variable ( "FA" ) { Set } , singleton ( variable ( "Y" ) { Int } ) ) ) , disjoint ( variable ( "FA" ) { Set } , singleton ( variable ( "Y" ) { Int } ) ) , \equals ( variable ( "Z" ) { Int } , select ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" ) { Int } ) ) , \equals ( variable ( "LENGTH" ) { Int } , plus ( variable ( "LA" ) { Int } , 1 ) ) , gt ( variable ( "Y" ) { Int } , 0 ) , .Patterns ) , \and ( listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" , 2 ) { Int } , variable ( "F" , 1 ) { Set } , variable ( "LENGTH" , 1 ) { Int } , .Patterns ) , \equals ( variable ( "LENGTH" , 1 ) { Int } , minus ( variable ( "LENGTH" ) { Int } , 1 ) ) , gt ( variable ( "Y" , 2 ) { Int } , 0 ) , \equals ( variable ( "Z" ) { Int } , select ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" , 2 ) { Int } ) ) , \equals ( variable ( "F" ) { Set } , union ( variable ( "F" , 1 ) { Set } , singleton ( variable ( "Y" , 2 ) { Int } ) ) ) , disjoint ( variable ( "F" , 1 ) { Set } , singleton ( variable ( "Y" , 2 ) { Int } ) ) , .Patterns ) )
    </k>
    <strategy>
      success
    </strategy>
    <trace>
         direct-proof
      ~> right-unfold-oneBody ( listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Z" ) { Int } , variable ( "F" ) { Set } , variable ( "LENGTH" ) { Int } , .Patterns ) , \and ( listSegmentRightLength ( variable ( "H" ) { ArrayIntInt } , variable ( "X" ) { Int } , variable ( "Y" , 2 ) { Int } , variable ( "F" , 1 ) { Set } , variable ( "LENGTH" , 1 ) { Int } , .Patterns ) , \equals ( variable ( "LENGTH" , 1 ) { Int } , minus ( variable ( "LENGTH" ) { Int } , 1 ) ) , gt ( variable ( "Y" , 2 ) { Int } , 0 ) , \equals ( variable ( "Z" ) { Int } , select ( variable ( "H" ) { ArrayIntInt } , variable ( "Y" , 2 ) { Int } ) ) , \equals ( variable ( "F" ) { Set } , union ( variable ( "F" , 1 ) { Set } , singleton ( variable ( "Y" , 2 ) { Int } ) ) ) , disjoint ( variable ( "F" , 1 ) { Set } , singleton ( variable ( "Y" , 2 ) { Int } ) ) , .Patterns ) )
    </trace>
  </goal>
</prover>
